app/services/affiliate-slots.js

```diff
@@ -56,10 +56,23 @@ const checkFilter = (filter, value) => (
     || (Array.isArray(filter) && (filter.length === 0 || filter.indexOf(value) > -1))
 );
 
+/**
+ *  Returns `true` if `launchOn` is undefined, empty or in the past
+ *
+ * @param {AffiliateUnit} unit
+ * @returns {boolean}
+ */
+const checkLaunchOn = (unit) => (
+  typeof unit.launchOn === 'undefined'
+    || (typeof unit.launchOn === 'string'
+      && (filter.length === 0 || (Date.parse(unit.launchOn) < Date.now()))
```

Comment:
Is `filter` a global var or should that be removed?

Reply:
Good catch